metpy/calc/kinematics.py

```diff
@@ -432,3 +433,109 @@ def geostrophic_wind(heights, f, dx, dy):
     grad = _gradient(heights, *deltas)
     dy, dx = grad[-2:]  # Throw away unused gradient components
     return -norm_factor * dy, norm_factor * dx
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]',
+             '[length]', '[length]', '[speed]', '[speed]')
+def storm_relative_helicity(u, v, p, srh_top, hgt, srh_bottom=0, storm_u=0 * units('m/s'),
+                            storm_v=0 * units('m/s'), dp=-1, exact=True):
+    r"""Calculate Storm Relative Helicity.
+
+    Needs u and v wind components, heights and pressures,
+    and top and bottom of SRH layer. An optional storm
+    motion vector can be specified. Partially adapted from
+    similar SharpPy code.
+
+    Parameters
+    ----------
+    srh_top : number
+        The height of the top of the desired layer for SRH.
+    srh_bottom : number
+        The height at the bottom of the SRH layer. Default is sfc.
+    hgts : array-like
+        The heights associatd with the data, provided in meters above mean
+        sea level and converted into meters AGL.
+    u : array-like
+        The u components of winds, same length as hgts
+    v : array-like
+        The u components of winds, same length as hgts
+    p : array-like
+        Pressure in hPa, same length as hgts
+    storm_u : number
+        u component of storm motion
+    storm_v : number
+        v component of storm motion
+    dp : negative integer
+        Pressure interval to interpolate the winds over.
+    exact: bool (optional, default = True)
+        switch between faster interpolated data and slower exact data
+    Returns
+    -------
+    number
+        p_srh : positive storm-relative helicity
+    number
+        n_srh : negative storm-relative helicity
+    number
+        t_srh : total storm-relative helicity
+
```

Comment:
Any thoughts for/against adding the typeset version? I think that's all that hasn't been addressed.